### PR TITLE
[zh] resync pod-lifecycle page

### PR DESCRIPTION
**Pull request by shawnhanx**

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/assign @tengqm

**Comment by tengqm** [comment]
Thank you for proposing this fix. However, we will not merge partial fixes as per stated in the [guideline](https://kubernetes.io/zh/docs/contribute/localization_zh/#%E6%9B%B4%E6%96%B0%E8%AF%91%E6%96%87). To find out what are the upstream changes to be synced, do this **on the main branch**:

```shell
./scripts/lsync.sh content/zh/docs/concepts/workloads/pods/pod-lifecycle.md
```

/hold

File: content/zh/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 1]
```diff
@@ -300,14 +300,14 @@ Pod 有一个 PodStatus 对象，其中包含一个
...
-  have started successfully.
+  have completed successfully.
 * `Ready`: the Pod is able to serve requests and should be added to the load
   balancing pools of all matching Services.
 -->
 * `PodScheduled`：Pod 已经被调度到某节点；
 * `ContainersReady`：Pod 中所有容器都已就绪；
 * `Initialized`：所有的 [Init 容器](/zh/docs/concepts/workloads/pods/init-containers/)
-  都已成功启动；
+  都已成功启动完成；
```
Comment (tengqm) [inline content/zh/docs/concepts/workloads/pods/pod-lifecycle.md:310]:
```suggestion
  都已成功完成；
```

File: content/zh/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 2]
```diff
@@ -425,93 +425,140 @@ When a Pod's containers are Ready but at least one custom condition is missing o
...
 
 <!--
-* [ExecAction](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#execaction-v1-core):
-  Executes a specified command inside the container. The diagnostic
-  is considered successful if the command exits with a status code of 0.
-
-* [TCPSocketAction](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#tcpsocketaction-v1-core):
-  Performs a TCP check against the Pod's IP address on
-  a specified port. The diagnostic is considered successful if the port is open.
+### Check mechanisms {#probe-check-methods}
```
Comment (tengqm) [inline content/zh/docs/concepts/workloads/pods/pod-lifecycle.md:441]: 这一小标题没翻译？

File: content/zh/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 3]
```diff
@@ -425,93 +425,140 @@ When a Pod's containers are Ready but at least one custom condition is missing o
...
 
-* [HTTPGetAction](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#httpgetaction-v1-core):
-  Performs an HTTP `GET` request against the Pod's IP
-  address on a specified port and path. The diagnostic is considered successful
-  if the response has a status code greater than or equal to 200 and less than 400.
--->
-- [ExecAction](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#execaction-v1-core)：
-  在容器内执行指定命令。如果命令退出时返回码为 0 则认为诊断成功。
+There are four different ways to check a container using a probe.
+Each probe must define exactly one of these four mechanisms:
```
Comment (tengqm) [inline content/zh/docs/concepts/workloads/pods/pod-lifecycle.md:444]: 这一段未翻译？

File: content/zh/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 4]
```diff
@@ -425,93 +425,140 @@ When a Pod's containers are Ready but at least one custom condition is missing o
...
+`exec`
+: 在容器内执行指定命令。如果命令退出时返回码为 0 则认为诊断成功。
+
+`grpc`
+: 使用 [gRPC](https://grpc.io/) 执行一个远程过程调用。
+  目标应该实现
+  [gRPC健康检查](https://grpc.io/grpc/core/md_doc_health-checking.html)。
+  如果响应的状态是 "SERVING"，则认为诊断成功。
+  gRPC 探针是一个 alpha 特性，只有在你启用了
+  启用 "GRPCContainerProbe" [featrue gate](/docs/reference/command-line-tools-reference/feature-gate/)时才能使用。
```
Comment (tengqm) [inline content/zh/docs/concepts/workloads/pods/pod-lifecycle.md:482]:
```suggestion
  "GRPCContainerProbe" [特性门控](/zh/docs/reference/command-line-tools-reference/feature-gate/)时才能使用。
```

File: content/zh/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 5]
```diff
@@ -425,93 +425,140 @@ When a Pod's containers are Ready but at least one custom condition is missing o
...
+
+`httpGet`
+: 对容器的 IP 地址上指定端口和路径执行 HTTP Get 请求。如果响应的状态码大于等于 200
```
Comment (tengqm) [inline content/zh/docs/concepts/workloads/pods/pod-lifecycle.md:485]:
```suggestion
: 对容器的 IP 地址上指定端口和路径执行 HTTP `GET` 请求。如果响应的状态码大于等于 200
```

File: content/zh/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 6]
```diff
@@ -425,93 +425,140 @@ When a Pod's containers are Ready but at least one custom condition is missing o
...
   且小于 400，则诊断被认为是成功的。
 
+`tcpSocket`
+: 对容器的 IP 地址上的指定端口执行 TCP 检查。如果端口打开，则诊断被认为是成功的。
+  如果远程系统（容器）在打开后立即关闭了，这算作是健康的。
```
Comment (tengqm) [inline content/zh/docs/concepts/workloads/pods/pod-lifecycle.md:490]:
```suggestion
  如果远程系统（容器）在打开l连接后立即将其关闭，这算作是健康的。
```

File: content/zh/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 7]
```diff
@@ -852,21 +899,20 @@ This avoids a resource leak as Pods are created and terminated over time.
...
+  the API reference documentation covering
+  [`.status`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodStatus) for Pod.
 -->
 * 动手实践[为容器生命周期时间关联处理程序](/zh/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/)。
 * 动手实践[配置存活态、就绪态和启动探针](/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/)。
 * 进一步了解[容器生命周期回调](/zh/docs/concepts/containers/container-lifecycle-hooks/)。
-* 关于 API 中定义的有关 Pod/容器的详细规范信息，
-  可参阅 [PodStatus](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#podstatus-v1-core)
-  和 [ContainerStatus](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#containerstatus-v1-core)。
+* 关于 API 中定义的有关 Pod和容器状态的详细规范信息，
```
Comment (tengqm) [inline content/zh/docs/concepts/workloads/pods/pod-lifecycle.md:916]:
```suggestion
* 关于 API 中定义的有关 Pod 和容器状态的详细规范信息，
```

File: content/zh/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 8]
```diff
@@ -852,21 +899,20 @@ This avoids a resource leak as Pods are created and terminated over time.
...
+  可参阅API参考文档[`.status`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodStatus)。
```
Comment (tengqm) [inline content/zh/docs/concepts/workloads/pods/pod-lifecycle.md:917]:
```suggestion
  可参阅 API 参考文档中 Pod 的 [`.status`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodStatus) 字段。
```

**Comment by shawnhanx** [comment]
@tengqm Please look again, Thanks.

**Comment by tengqm** [comment]
/retitle [zh] resync pod-lifecycle page
/approve

Thanks.

**Pull request merged**